advanced-list/src/app.js

```diff
@@ -50,30 +45,23 @@ function view(children$, name = '') {
   );
 }
 
-const TickerWrapper = sources => function TickerComponent(color$, id) {
-  const ticker = isolate(Ticker)({DOM: sources.DOM, color: color$});
-  return {
-    DOM: ticker.DOM,
-    action$: ticker.action$.map(a => {
-      a.id = id;
-      return a;
-    }),
-  };
-}
-
 function App(sources) {
-  const tickerProxyAction$ = xs.create();
-  const action$ = intent(sources.DOM, tickerProxyAction$);
-  const tickers$ = model(action$, TickerWrapper(sources));
-  const tickerViews$ = tickers$.map(list => list.map(t => t.DOM));
-  const tickerAction$ = tickers$
-    .map(list => xs.merge(...list.map(t => t.action$)))
-    .flatten();
-  tickerProxyAction$.imitate(tickerAction$);
+  const tickers = Collection(Ticker, sources, {
+    remove$ (tickers, ticker) {
+      return tickers.remove(ticker);
+    }
+  });
```

Comment:
You're on the money. The reasoning goes a little something like this.

`tickers` is a collection of `Ticker` components. It's immutable, because in Cycle values that change are streams.

`tickers$` is a stream of `tickers`, changing over time. It's a common and useful pattern to build that stream by `fold`ing over a stream of `reducers`.

Sometimes a component in a collection needs to send a message to it's parent. The most common example is a `remove` button. However, taking the sinks of each component, wrapping it with an id and setting up a circular dependency takes a lot of boilerplate and adds a lot of syntactic noise to the application.

Instead we provide sink handlers to `Collection`, which as you've noted map item sinks into `reducers`, so that they can be used in the creation of `tickers$`. Sink handlers are passed `(state, item, sinkEvent)` but are then wrapped so that they're called as `(state) => state`. This means the user is saved the trouble of setting up ids or isolation.